src/tests/Register.test.js

```diff
@@ -0,0 +1,257 @@
+    });
+
+    fireEvent.change(passwordInput, {
+      target: { value: 'Testpassword123!' },
+    });
+
+    fireEvent.change(confirmPasswordInput, {
+      target: { value: 'Testpassword123!' },
+    });
+
+    expect(registerBtn).toBeDisabled();
+  });
+
+  it('Register button should be disabled if the email field is empty', () => {
+    renderWithReduxAndRouter(Register, {
+      route,
+    });
+    const usernameInput = screen.getByTestId('username').querySelector('input');
+    const emailInput = screen.getByTestId('email').querySelector('input');
+    const passwordInput = screen.getByTestId('password').querySelector('input');
+    const confirmPasswordInput = screen
+      .getByTestId('confirm_password')
+      .querySelector('input');
+    const registerBtn = screen.getByTestId('register_btn');
+
+    fireEvent.change(usernameInput, {
+      target: { value: 'username123' },
+    });
+
+    fireEvent.change(emailInput, {
+      target: { value: '' },
+    });
+
+    fireEvent.change(passwordInput, {
+      target: { value: 'Testpassword123!' },
+    });
+
+    fireEvent.change(confirmPasswordInput, {
+      target: { value: 'Testpassword123!' },
+    });
+
+    expect(registerBtn).toBeDisabled();
+  });
+});
+
+describe('Password Eye Icon:', () => {
+  it('Clicking password eye icon of the password input should display the password text', () => {
+    renderWithReduxAndRouter(Register, {
+      route,
+    });
+    const passwordInput = screen.getByTestId('password').querySelector('input');
+    const passwordDisplayHideIcon = screen.getByTestId('hide-display-password');
+    expect(passwordInput).toHaveAttribute('type', 'password');
+    fireEvent.click(passwordDisplayHideIcon);
+    expect(passwordInput).toHaveAttribute('type', 'text');
+  });
+
+  it('Clicking password eye icon of the confirm password input should display the password text', () => {
+    renderWithReduxAndRouter(Register, {
+      route,
+    });
+    const passwordInput = screen
+      .getByTestId('confirm_password')
+      .querySelector('input');
+    const passwordDisplayHideIcon = screen.getByTestId(
+      'hide-display-confirm-password'
+    );
+    expect(passwordInput).toHaveAttribute('type', 'password');
+    fireEvent.click(passwordDisplayHideIcon);
+    expect(passwordInput).toHaveAttribute('type', 'text');
+  });
+});
+
+describe('Successful Registration', () => {
+  it('Should see success message', async () => {
+    axios.post.mockImplementationOnce(() =>
+      Promise.resolve({
+        data: { data: ['Your email is confirmed!'] },
+        status: 200,
+      })
+    );
+
+    const callback = jest.fn(() => Promise.resolve(true));
+    const registration = {
+      username: 'username123',
+      email: 'test@gmail.com',
+      password: 'Testpassword123!',
+      confirm_password: 'Testpassword123!',
+    };
+
+    const result = postRegister((registration, callback));
+    console.log(result);
+  });
+});
```

Comment:
@jennypoon You can change this test to this (I have tested it):

```suggestion
describe('Successful Registration', () => {
  it('Should see success message', async () => {
    const response = {
      data: { detail: 'Please confirm your email to Login succesfully' },
      status: 200,
    };

    axios.post.mockImplementationOnce(() => Promise.resolve(response));

    const callback = jest.fn();
    const dispatchMock = jest.fn();
    const registration = {
      username: 'username123',
      email: 'test@gmail.com',
      password: 'Testpassword123!',
      confirm_password: 'Testpassword123!',
    };

    const dispatch = postRegister(registration, callback);
    await dispatch(dispatchMock);
    expect(callback).toHaveBeenCalled();
    expect(dispatchMock).toHaveBeenCalledWith({
      payload: response.data,
      type: 'REGISTER',
    });
  });
});
```